File: stvs/src/test/java/edu/kit/iti/formal/stvs/view/spec/timingdiagram/renderer/VerticalResizeContainerControllerDemo.java
package edu.kit.iti.formal.stvs.view.spec.timingdiagram.renderer;

import edu.kit.iti.formal.stvs.Demo;
import edu.kit.iti.formal.stvs.view.JavaFxTest;
import javafx.application.Application;
import javafx.scene.Scene;
import javafx.scene.layout.Pane;
import javafx.scene.layout.Priority;
import javafx.scene.layout.VBox;
import javafx.scene.paint.Color;
import javafx.scene.shape.Rectangle;
import org.junit.Test;
import org.junit.experimental.categories.Category;

/**
 * Created by leonk on 10.02.2017.
 */
@Category(Demo.class)
public class VerticalResizeContainerControllerDemo {

  @Test
  public void javaFxTest() {
    JavaFxTest.setToBeViewed(this::simpleScene);
    Application.launch(JavaFxTest.class);
  }

  private Scene simpleScene() {
    try {
      VBox root = new VBox();
      Pane pane = new Pane();
      Rectangle rectangle = new Rectangle(100, 100, Color.AQUA);
      pane.getChildren().addAll(rectangle);
      VerticalResizeContainerController controller = new VerticalResizeContainerController(pane);
      root.getChildren().addAll(controller.getView());
      VBox.setVgrow(controller.getView(), Priority.ALWAYS);
      Scene scene = new Scene(root, 800, 600);
      return scene;
    } catch (Exception e) {
      e.printStackTrace();
      return null;
    }
  }

}

File: stvs/src/test/kotlin/edu/kit/iti/formal/stvs/view/spec/timingdiagram/renderer/VerticalResizeContainerControllerDemo.kt
/* *****************************************************************
 * This file belongs to verifaps-lib (https://verifaps.github.io).
 * SPDX-License-Header: GPL-3.0-or-later
 * 
 * This program isType free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program isType distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a clone of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * *****************************************************************/
package edu.kit.iti.formal.stvs.view.spec.timingdiagram.renderer

import edu.kit.iti.formal.stvs.view.JavaFxTest
import javafx.application.Application
import javafx.scene.Scene
import javafx.scene.layout.Pane
import javafx.scene.layout.Priority
import javafx.scene.layout.VBox
import javafx.scene.paint.Color
import javafx.scene.shape.Rectangle
import org.junit.jupiter.api.Tag
import org.junit.jupiter.api.Test
import java.util.function.Supplier

/**
 * Created by leonk on 10.02.2017.
 */
@Tag("demo")
class VerticalResizeContainerControllerDemo {
    @Test
    fun javaFxTest() {
        JavaFxTest.Companion.setToBeViewed(Supplier<Scene?> { this.simpleScene() })
        Application.launch(JavaFxTest::class.java)
    }

    private fun simpleScene(): Scene? {
        try {
            val root = VBox()
            val pane = Pane()
            val rectangle = Rectangle(100.0, 100.0, Color.AQUA)
            pane.children.addAll(rectangle)
            val controller = VerticalResizeContainerController(pane)
            root.children.addAll(controller.view)
            VBox.setVgrow(controller.view, Priority.ALWAYS)
            val scene = Scene(root, 800.0, 600.0)
            return scene
        } catch (e: Exception) {
            e.printStackTrace()
            return null
        }
    }
}